File: redsheet-portal/src/app/database-models/nano-project-negotiable.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectModule, vNanoProjectNegotiableRecord } from './'
export class NanoProjectNegotiable {
    constructor(nanoProjectNegotiable?: NanoProjectNegotiable | vNanoProjectNegotiableRecord) {
        let properties = ["ldo", "mdo", "nanoProjectNegotiableId", "requirement", "projectModuleId",];
        for (let property of properties)
            if (nanoProjectNegotiable && nanoProjectNegotiable[property])
                this[property] = nanoProjectNegotiable[property];
    }

    @required()
    @maxLength(150)
    ldo: string = undefined;

    @required()
    @maxLength(150)
    mdo: string = undefined;

    nanoProjectNegotiableId: number = 0;

    @required()
    @maxLength(150)
    requirement: string = undefined;

    @range(0, 2147483647)
    projectModuleId: number = undefined;
    projectModule: ProjectModule;


}

File: RedSheet.DbEntities/Models/NanoProjectNegotiable.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("NanoProjectNegotiables", Schema = "dbo")]
    [Serializable]
    public partial class NanoProjectNegotiable
    {

        [Required]
        [MaxLength(150)]
        public string ldo { get; set; }

        [Required]
        [MaxLength(150)]
        public string mdo { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int NanoProjectNegotiableId { get; set; }

        [Required]
        [MaxLength(150)]
        public string Requirement { get; set; }

        [RelationshipTableAttribue("ProjectModules", "dbo")]
        //Relationships
        public int ProjectModuleId { get; set; }

        [ForeignKey("ProjectModuleId")]
        public ProjectModule ProjectModule { get; set; }

        public NanoProjectNegotiable()
        {
        }

    }
}

